src/streaming/controllers/EventController.js

```diff
@@ -216,7 +221,9 @@ function EventController() {
 
                 if (curr !== undefined) {
                     presentationTime = curr.presentationTime / curr.eventStream.timescale;
-                    if (presentationTime === 0 || (presentationTime <= currentVideoTime && presentationTime + presentationTimeThreshold > currentVideoTime)) {
+                    const period = curr.eventStream.period;
+                    const start = (period && typeof period.start === 'number') ? period.start : presentationTimeThreshold;
+                    if (presentationTime === 0 || presentationTime + start <= currentVideoTime) {
```

Comment:
Perform calculations based on the start of the period instead of the current playhead time.